platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/StackTraceLocation.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import org.gradle.api.problems.FileLocation;
+import org.gradle.api.problems.ProblemLocation;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * A stack trace for the location of a problem.
+ * <p>
+ * For example, the stacktrace of a direct deprecation warning.
+ */
+public interface StackTraceLocation extends ProblemLocation {
+
+    /**
+     * The source file location inferred from the stacktrace.
+     * <p>
+     * Most of the time, this is a location in a build file.
+     * {@code null} if the source file location could not be inferred.
+     */
+    @Nullable
+    FileLocation getFileLocation();
+
+    /**
+     * The stack trace elements that lead to the problem.
+     */
+    List<StackTraceElement> getStackTrace();
```

Comment:
🤔  I'm not sure we should use `StackTraceElement` as-is, because of the revised contract in Java 9. I understand this is an internal API so far, but maybe we should at least add a TODO to mark that we probably want to have some wrapper similar to `StackTraceElementPlaceholder`.

Reply:
For Develocity, `StackTraceElement` is just fine. Do we expect to transmit stacktrace elements between JVMs with different versions?

Reply:
So far we did not plan that, but this new location type opens up the opportunity.

Reply:
If we use Java Serialization, I suppose this will blow up because the serial version UUID is different, right? Though we don't use Java Serialization, right?

Reply:
What we do on the Develocity side is that we ignore module name and version completely. That way the version of Java doesn't matter.

Reply:
Can you create an issue for this? I'll move ahead now with the current changes. We can always change the internal type.